frontend/javascripts/oxalis/model/bucket_data_handling/mappings.ts

```diff
@@ -56,58 +86,72 @@ class Mappings {
     );
   }
 
+  is64Bit() {
+    const elementClass = getElementClass(Store.getState().dataset, this.layerName);
+    return elementClass === "uint64" || elementClass === "int64";
+  }
+
   async updateMappingTextures(mapping: Mapping | null | undefined): Promise<void> {
     if (mapping == null) return;
-    console.time("Time to create mapping texture");
-    const mappingSize = mapping.size;
-    // The typed arrays need to be padded with 0s so that their length is a multiple of MAPPING_TEXTURE_WIDTH
-    const paddedLength =
-      mappingSize + MAPPING_TEXTURE_WIDTH - (mappingSize % MAPPING_TEXTURE_WIDTH);
-    const keys = new Uint32Array(paddedLength);
-    const values = new Uint32Array(paddedLength);
-    const mappingKeys = Array.from(mapping.keys());
-    mappingKeys.sort((a, b) => a - b);
-    keys.set(mappingKeys);
-    // @ts-ignore mappingKeys are guaranteed to exist in mapping as they are mapping.keys()
-    values.set(mappingKeys.map((key) => mapping.get(key)));
-    // Instantiate the Uint8Arrays with the array buffer from the Uint32Arrays, so that each 32-bit value is converted
-    // to four 8-bit values correctly
-    const uint8Keys = new Uint8Array(keys.buffer);
-    const uint8Values = new Uint8Array(values.buffer);
-    console.timeEnd("Time to create mapping texture");
-
-    if (mappingSize > MAPPING_TEXTURE_WIDTH ** 2) {
-      throw new Error(messages["mapping.too_big"]);
+    if (this.cuckooTable == null) {
+      throw new Error("cuckooTable null when updateMappingTextures was called.");
     }
 
-    this.mappingLookupTexture.update(
-      uint8Keys,
-      0,
-      0,
-      MAPPING_TEXTURE_WIDTH,
-      uint8Keys.length / MAPPING_TEXTURE_WIDTH / 4,
-    );
-    this.mappingTexture.update(
-      uint8Values,
-      0,
-      0,
-      MAPPING_TEXTURE_WIDTH,
-      uint8Values.length / MAPPING_TEXTURE_WIDTH / 4,
-    );
+    const { changed, onlyA, onlyB } =
+      this.previousMapping != null
+        ? cachedDiffMappings(this.previousMapping, mapping)
```

Comment:
I think my main motivation for this code was the ability to have a `setCacheResultForDiffMappings` function. The mapping is usually changed frequently because of newly arriving buckets. At that point, we know exactly which ids will be added and which will be deleted (because this is how we mutate the mapping). Afterwards, the differ would compute exactly the same infos. To avoid this, `setCacheResultForDiffMappings` is used. The performance advantage should be quite high due to the frequency (especially for large mappings).
Theoretically, one could also use `setCacheResultForDiffMappings` when doing proofreading actions. However, such actions are rarer and I don't know whether it's really a bottleneck.

Reply:
Ah, now I get why this makes a lot of sense, thanks for explaining!